File: 백준/Silver/21921. 블로그/블로그.py
import sys

n, x = map(int, sys.stdin.readline().split())
people = list(map(int, sys.stdin.readline().split()))
max_people = sum(people[:x])
current_people = max_people
cnt = 1

for i in range(1, n-x+1):
    current_people = current_people - people[i-1] + people[i+x-1]
    if current_people > max_people:
        max_people = current_people
        cnt = 1
    elif current_people == max_people:
        cnt += 1

if max_people != 0:
    sys.stdout.write(f"{max_people}\n")
    sys.stdout.write(f"{cnt}\n")
else:
    sys.stdout.write("SAD\n")

File: 백준/Silver/21921. 블로그/블로그.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    
    int n, x;
    cin >> n >> x;
    
    vector<int> people(n);
    for (int i = 0; i < n; i++)
        cin >> people[i];

    int max_people = 0;
    for (int i = 0; i < x; i++)
        max_people += people[i];

    int current_sum = max_people;
    int cnt = 1;

    for (int i = 1; i <= n-x; i++) {
        current_sum = current_sum - people[i-1] + people[i+x-1];

        if (current_sum > max_people) {
            max_people = current_sum;
            cnt = 1;
        }
        else if (current_sum == max_people)
            cnt++;
    }

    if (max_people != 0) {
        cout << max_people << "\n";
        cout << cnt << "\n";
    }
    else
        cout << "SAD" << "\n";

    return 0;
}